db.py

```diff
@@ -40,14 +41,14 @@ def validate_active(self, key, value):
         return value
 
     def allowed(self, access_level):
-        if access_level == Role.ADMIN:
-            return self.admin
-        elif access_level == Role.RESEARCHER:
-            return self.is_researcher
-        elif access_level == Role.TEACHER:
-            return self.is_teacher
-        else:
-            return False
+        role_access = {
+            Role.USER: True,
```

Comment:
You can therefore remove this one and let `False` be the default value. Changing this to a dictionary lookup remains a good idea.